ports/cpp/source/antlr4-c3/CodeCompletionCore.cpp

```diff
@@ -85,25 +85,37 @@ CodeCompletionCore::CodeCompletionCore(antlr4::Parser* parser)
     , atn(parser->getATN())
     , vocabulary(parser->getVocabulary())
     , ruleNames(parser->getRuleNames())
-    , timeoutMS(0)
+    , timeout(0)
     , cancel(nullptr) {
 }
 
-CandidatesCollection CodeCompletionCore::collectCandidates(  // NOLINT
-    size_t caretTokenIndex,
-    antlr4::ParserRuleContext* context,
-    size_t timeoutMS,
-    std::atomic<bool>* cancel
+void CodeCompletionCore::setIgnoredTokens(std::unordered_set<size_t>&& ignoredTokens) {
+  this->ignoredTokens = std::move(ignoredTokens);
+}
+
+void CodeCompletionCore::setPreferredRules(std::unordered_set<size_t>&& preferredRules) {
+  this->preferredRules = std::move(preferredRules);
+}
+
+void CodeCompletionCore::setTranslateRulesTopDown(bool isEnabled) {
+  this->translateRulesTopDown = isEnabled;
+}
```

Comment:
What's the purpose of these setters? Since there's no side processing when setting a value, there's no need to use a setter.